### PR TITLE
Update `AddChannel`, `AsChannelFirst` with `EnsureChannelFirst`

### LINKED ISSUE #517: review deprecated _meta_dict api usage
part of https://github.com/Project-MONAI/MONAI/issues/7018 we recently set `image_only=True` in `LoadImage`, so the logic based on _meta_dict might not work any more, e.g. https://github.com/Project-MONAI/model-zoo/blob/a9acb18ad12af9f1829402a806300a5a5f917371/models/spleen_ct_segmentation/configs/evaluate.json#L71

### LINKED ISSUE #7036: test_dataset_tracking
```
[2023-09-22T18:59:24.123Z] ======================================================================
[2023-09-22T18:59:24.123Z] ERROR: test_dataset_tracking (tests.test_handler_mlflow.TestHandlerMLFlow)
[2023-09-22T18:59:24.123Z] ----------------------------------------------------------------------
[2023-09-22T18:59:24.123Z] Traceback (most recent call last):
[2023-09-22T18:59:24.123Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/config_item.py", line 293, in instantiate
[2023-09-22T18:59:24.123Z]     return instantiate(modname, mode, **args)
[2023-09-22T18:59:24.123Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/utils/module.py", line 248, in instantiate
[2023-09-22T18:59:24.123Z]     raise ModuleNotFoundError(f"Cannot locate class or function path: '{__path}'.")
[2023-09-22T18:59:24.123Z] ModuleNotFoundError: Cannot locate class or function path: 'AsChannelFirstd'.
[2023-09-22T18:59:24.123Z] 
[2023-09-22T18:59:24.123Z] The above exception was the direct cause of the following exception:
[2023-09-22T18:59:24.123Z] 
[2023-09-22T18:59:24.123Z] Traceback (most recent call last):
[2023-09-22T18:59:24.123Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/test_handler_mlflow.py", line 271, in test_dataset_tracking
[2023-09-22T18:59:24.123Z]     infer_dataset = workflow.dataset
[2023-09-22T18:59:24.123Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/workflows.py", line 129, in __getattr__
[2023-09-22T18:59:24.123Z]     return self._get_property(name=name, property=self.properties[name])
[2023-09-22T18:59:24.123Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/workflows.py", line 365, in _get_property
[2023-09-22T18:59:24.123Z]     return self.parser.get_parsed_content(id=prop_id) if prop_id is not None else None
[2023-09-22T18:59:24.124Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/config_parser.py", line 290, in get_parsed_content
[2023-09-22T18:59:24.124Z]     return self.ref_resolver.get_resolved_content(id=id, **kwargs)
[2023-09-22T18:59:24.124Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/reference_resolver.py", line 193, in get_resolved_content
[2023-09-22T18:59:24.124Z]     return self._resolve_one_item(id=id, **kwargs)
[2023-09-22T18:59:24.124Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/reference_resolver.py", line 163, in _resolve_one_item
[2023-09-22T18:59:24.124Z]     self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
[2023-09-22T18:59:24.124Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/reference_resolver.py", line 163, in _resolve_one_item
[2023-09-22T18:59:24.124Z]     self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
[2023-09-22T18:59:24.124Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/reference_resolver.py", line 171, in _resolve_one_item
[2023-09-22T18:59:24.124Z]     self.resolved_content[id] = item.instantiate() if kwargs.get("instantiate", True) else item
[2023-09-22T18:59:24.124Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/bundle/config_item.py", line 295, in instantiate
[2023-09-22T18:59:24.124Z]     raise RuntimeError(f"Failed to instantiate {self}") from e
[2023-09-22T18:59:24.124Z] RuntimeError: Failed to instantiate ConfigComponent: 
[2023-09-22T18:59:24.124Z] {'_target_': 'AsChannelFirstd', 'keys': ['image']}
[2023-09-22T18:59:24.124Z] 
[2023-09-22T18:59:24.124Z] ----------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes https://github.com/Project-MONAI/MONAI/issues/7036.
Fixes https://github.com/Project-MONAI/model-zoo/issues/517.

### Description
- Update `AddChannel`, `AsChannelFirst` with `EnsureChannelFirst`.
- Update `workflow` to `workflow_type`
- Review deprecated _meta_dict API usage

### Status
**Work in progress**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Comment by KumoLiu** [comment]
Update all related bundles including `AddChannel` and `AsChannelFirst` and verify them except for "ventricular_short_axis_3label" since I didn't have suitable data to test. cc @ericspod, could you please help update this one and verify it? Thanks!

**Comment by ericspod** [comment]
Hi @KumoLiu how much data did you need? I can save the output for the example (256, 256) 2D image if that's enough.

**Comment by KumoLiu** [comment]
> Hi @KumoLiu how much data did you need? I can save the output for the example (256, 256) 2D image if that's enough.

Hi @ericspod, yeah, that would be great, just one test data with the same format used in the bundle is okay. Thanks in advance!

**Comment by ericspod** [comment]
[SC-N-2-3-0_seg.zip](https://github.com/Project-MONAI/model-zoo/files/12730547/SC-N-2-3-0_seg.zip)
This is the output of the network converted to uint8 labels. This follows the notebook in the `docs` directory and should be enough to test the output of the network.

[SC-N-2-3-0_pred.zip](https://github.com/Project-MONAI/model-zoo/files/12730560/SC-N-2-3-0_pred.zip)
This is the raw output tensor from the network if you want to use this instead.

**Comment by KumoLiu** [comment]
I may leave "ventricular_short_axis_3label" this bundle in this PR.
It still using the API before v0.6.
https://github.com/Project-MONAI/model-zoo/blob/a9acb18ad12af9f1829402a806300a5a5f917371/models/ventricular_short_axis_3label/configs/train.json#L102

Not easy to update this one, when I remove the `as_tensor_output`, `AddChannel` and specify `data_type` in `EnsureType`, it throws the error below.

Update: fixed
<details>error message
opt/pytorch/pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:365: operator(): block: [64,0,0], thread: [63,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
2023-10-11 03:38:01,405 - ignite.engine.engine.SupervisedTrainer - ERROR - Current run is terminating due to exception: CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

2023-10-11 03:38:01,430 - ignite.engine.engine.SupervisedTrainer - ERROR - Engine run is terminating due to exception: CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

2023-10-11 03:38:01,431 - ignite.engine.engine.SupervisedTrainer - INFO - Deleted previous saved final checkpoint: model_final_iteration=1.pt
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/ignite/engine/engine.py", line 1068, in _run_once_on_dataset_as_gen
    self.state.output = self._process_function(self, self.state.batch)
  File "/workspace/Code/MONAI/monai/engines/trainer.py", line 230, in _iteration
    _compute_pred_loss()
  File "/workspace/Code/MONAI/monai/engines/trainer.py", line 216, in _compute_pred_loss
    engine.state.output[Keys.LOSS] = engine.loss_function(engine.state.output[Keys.PRED], targets).mean()
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/workspace/Code/MONAI/monai/losses/dice.py", line 176, in forward
    intersection = torch.sum(target * input, dim=reduce_axis)
  File "/workspace/Code/MONAI/monai/data/meta_tensor.py", line 282, in __torch_function__
    ret = super().__torch_function__(func, types, args, kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/_tensor.py", line 1296, in __torch_function__
    ret = func(*args, **kwargs)
RuntimeError: CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

**Comment by ericspod** [comment]
> I may leave "ventricular_short_axis_3label" this bundle in this PR. It still using the API before v0.6.
> 
> https://github.com/Project-MONAI/model-zoo/blob/a9acb18ad12af9f1829402a806300a5a5f917371/models/ventricular_short_axis_3label/configs/train.json#L102

Is this totally fixed now? I think the error you have is from some issue on the platform side or something with the versions of Pytorch and/or CUDA. I don't remember what the `as_tensor_output` argument was meant to fix, it might have been MetaTensor related.

**Comment by KumoLiu** [comment]
> > I may leave "ventricular_short_axis_3label" this bundle in this PR. It still using the API before v0.6.
> > https://github.com/Project-MONAI/model-zoo/blob/a9acb18ad12af9f1829402a806300a5a5f917371/models/ventricular_short_axis_3label/configs/train.json#L102
> 
> Is this totally fixed now? I think the error you have is from some issue on the platform side or something with the versions of Pytorch and/or CUDA. I don't remember what the `as_tensor_output` argument was meant to fix, it might have been MetaTensor related.

Yes, it works now. Thanks!

File: models/breast_density_classification/configs/metadata.json [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
         "0.1.4": "Changed Readme",
         "0.1.3": "Change input_dim from 229 to 299",
         "0.1.2": "black autofix format and add name tag",
         "0.1.1": "update license files",
         "0.1.0": "complete the model package"
     },
-    "monai_version": "1.0.0",
-    "pytorch_version": "1.12.1",
-    "numpy_version": "1.21.2",
+    "monai_version": "865972f7a791bf7b42efbcd87c8402bd865b329e",
```
Comment (yiheng-wang-nv) [inline models/breast_density_classification/configs/metadata.json]: Hi @KumoLiu @wyli , do you think we should hold the PR until we have a new monai tag that covers this commit?

Comment (wyli) [reply models/breast_density_classification/configs/metadata.json]: yes, this is for temp test only

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**